subprojects/gst-plugins-good/gst/rtp/gstrtph266pay.c

```diff
@@ -0,0 +1,792 @@
+        "clock-rate = (int) 90000, " "encoding-name = (string) \"H266\"")
+    );
+
+#define DEFAULT_CONFIG_INTERVAL 0
+
+enum
+{
+  PROP_0,
+  PROP_CONFIG_INTERVAL,
+};
+
+static void gst_rtp_h266_pay_finalize (GObject * object);
+static void gst_rtp_h266_pay_set_property (GObject * object, guint prop_id,
+    const GValue * value, GParamSpec * pspec);
+static void gst_rtp_h266_pay_get_property (GObject * object, guint prop_id,
+    GValue * value, GParamSpec * pspec);
+
+static GstStateChangeReturn gst_rtp_h266_pay_change_state (GstElement *
+    element, GstStateChange transition);
+
+static gboolean gst_rtp_h266_pay_set_caps (GstRTPBasePayload * rtpbasepay,
+    GstCaps * caps);
+static GstFlowReturn gst_rtp_h266_pay_handle_buffer (GstRTPBasePayload *
+    rtpbasepay, GstBuffer * buffer);
+static gboolean gst_rtp_h266_pay_sink_event (GstRTPBasePayload * rtpbasepay,
+    GstEvent * event);
+
+static void gst_rtp_h266_pay_process_nalu (GstRtpH266Pay * rtph266pay,
+    GstBuffer * nalu_buf);
+static void gst_rtp_h266_pay_update_xps_cache (GstRtpH266Pay * rtph266pay,
+    const Nalu * nalu);
+static void gst_rtp_h266_pay_clear_xps_cache (GstRtpH266Pay * rtph266pay);
+static void gst_rtp_h266_pay_queue_xps_cache (GstRtpH266Pay * rtph266pay);
+static gboolean gst_rtp_h266_pay_has_to_send_xps (GstRtpH266Pay * rtph266pay,
+    const Nalu * nalu);
+static void gst_rtp_h266_pay_set_au_boundaries (GstRtpH266Pay * rtph266pay);
+static GstFlowReturn gst_rtp_h266_pay_push_pending_data (GstRtpH266Pay *
+    rtph266pay, gboolean eos);
+static GstFlowReturn gst_rtp_h266_pay_push_up (GstRtpH266Pay * rtph266pay,
+    const Nalu * nalu);
+static GstFlowReturn gst_rtp_h266_pay_push_fu (GstRtpH266Pay * rtph266pay,
+    const Nalu * nalu);
+//static GstFlowReturn gst_rtp_h266_pay_push_ap (GstRtpH266Pay * rtph266pay, GQueue *nalus);
+static gboolean gst_rtp_h266_pay_up_fits_in_mtu (const GstRtpH266Pay *
+    rtph266pay, const Nalu * nalu);
+static GstBuffer *gst_rtp_h266_pay_extract_fu (GstRtpH266Pay * rtph266pay,
+    const Nalu * nalu, gsize offset, gsize size, gboolean last);
+
+static Nalu *nalu_new (GstBuffer * nalu_buf);
+static void nalu_free (Nalu * nalu);
+static Nalu *nalu_copy (const Nalu * nalu);
+
+static void
+gst_rtp_h266_pay_class_init (GstRtpH266PayClass * klass)
+{
+  GObjectClass *gobject_class;
+  GstElementClass *gstelement_class;
+  GstRTPBasePayloadClass *gstrtpbasepayload_class;
+
+  gobject_class = (GObjectClass *) klass;
+  gstelement_class = (GstElementClass *) klass;
+  gstrtpbasepayload_class = (GstRTPBasePayloadClass *) klass;
+
+  gobject_class->finalize = GST_DEBUG_FUNCPTR (gst_rtp_h266_pay_finalize);
+  gobject_class->set_property =
+      GST_DEBUG_FUNCPTR (gst_rtp_h266_pay_set_property);
+  gobject_class->get_property =
+      GST_DEBUG_FUNCPTR (gst_rtp_h266_pay_get_property);
+
+  g_object_class_install_property (gobject_class,
+      PROP_CONFIG_INTERVAL,
+      g_param_spec_int ("config-interval",
```

Comment:
Since the property is not yet handled in set/get properties, should we either remove/comment/disable this property for now or implement the set/get property?

Reply:
We won't do a MR to the community until this is finished...

Reply:
I agree with removing it. We can add the property later.

Reply:
Done. I've moved everything related to config-interval into https://github.com/fluendo/gstreamer/pull/42